core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -907,11 +918,21 @@ class ReplicaManager(val config: KafkaConfig,
       }
     }
 
-    // Wrap the given callback function with another function that will update the HW for the remote follower
-    def maybeUpdateHwAndSendResponse(fetchPartitionData: Seq[(TopicPartition, FetchPartitionData)]): Unit = {
+    // Wrap the given callback function with another function that will update the HW and partition reassignment lag
+    // for the remote follower
+    def updateReplicaAndSendResponse(fetchPartitionData: Seq[(TopicPartition, FetchPartitionData)]): Unit = {
       if (isFromFollower) {
         fetchPartitionData.foreach {
-          case (tp, partitionData) => updateFollowerHighWatermark(tp, replicaId, partitionData.highWatermark)
+          case (tp, partitionData) =>
+            val replicaOpt = nonOfflinePartition(tp).flatMap(_.getReplica(replicaId))
+            replicaOpt match {
+              case Some(replica) =>
+                replica.updateLastSentHighWatermark(partitionData.highWatermark)
+                replica.updateReassignmentLag(partitionData.messageOffset, isAddingReplica(tp, replicaId))
```

Comment:
I'll be frank - I'm a bit lost in the code. Could you help clarify how `partitionData.messageOffset` is different than `_logEndOffsetMetadata.messageOffset` at this point? I believe we update the latter in  line 902 - `if (isFromFollower) updateFollowerFetchState(replicaId, result)`

Reply:
Looking at it more deeply it's not different. I think I'll use that one and compute reassignment lag on the fly.